lib/policy_machine_storage_adapters/active_record.rb

```diff
@@ -663,22 +633,32 @@ def transaction(&block)
     # Returns true if the user has the operation on the object
     def is_privilege?(user_or_attribute, operation, object_or_attribute)
       policy_classes_containing_object = policy_classes_for_object_attribute(object_or_attribute)
+      operation_id = operation.try(:unique_identifier) || operation.to_s
+
       if policy_classes_containing_object.count < 2
-        is_privilege_single_policy_class(user_or_attribute, operation, object_or_attribute)
+        accessible_operations(user_or_attribute, object_or_attribute, operation_id).any?
       else
-        is_privilege_multiple_policy_classes(user_or_attribute, operation, object_or_attribute, policy_classes_containing_object)
+        policy_classes_containing_object.all? do |policy_class|
+          accessible_operations(user_or_attribute, object_or_attribute, operation_id).any?
+        end
       end
     end
 
     ## Optimized version of PolicyMachine#scoped_privileges
     # Returns all operations the user has on the object
     def scoped_privileges(user_or_attribute, object_or_attribute, options = {})
       policy_classes_containing_object = policy_classes_for_object_attribute(object_or_attribute)
-      if policy_classes_containing_object.count < 2
-        scoped_privileges_single_policy_class(user_or_attribute, object_or_attribute, options)
-      else
-        scoped_privileges_multiple_policy_classes(user_or_attribute, object_or_attribute, policy_classes_containing_object, options)
-      end
+
+      operations =
+        if policy_classes_containing_object.count < 2
+          accessible_operations(user_or_attribute, object_or_attribute)
+        else
+          policy_classes_containing_object.flat_map do |policy_class|
+            accessible_operations(user_or_attribute, policy_class.ancestors)
+          end
+        end
+
+      options[:order] ? operations.sort : operations
```

Comment:
I'm not sure what the default order is, but refactoring has it limits 😝 . That line was around before this PR, and for now, I'm not sure changing it here is appropriate. We may eventually want to sort by the unique identifier, but I'm not sure. For a later day, unless you feel passionately about it.